connectors/sources/sharepoint.py

```diff
@@ -111,6 +112,10 @@ def __init__(self, configuration):
         self.access_token = None
         self.token_expires_at = None
 
+        self.limiter = AsyncLimiter(
```

Comment:
Should these values be "advanced" configurable fields? We could fall back to these specific values as default, if none are configured.

Reply:
We can add this in configurable fields but what if user entered value more than rate limit value?

Reply:
I think it's possible to catch that with the `constraint` field via declarative configuration, right? So you basically specify an upper bound constraint in `get_default_configuration`